sources.go

```diff
@@ -34,6 +34,7 @@ func Exec(s string) *Pipe {
 // status will be set, and if the file does exist, the pipe will have no error
 // status.
 func IfExists(filename string) *Pipe {
+	filename = strings.TrimRight(filename, "\n")
```

Comment:
Do we need this? If the intention of this PR is only to remove trailing newlines from strings that come from a pipe (for example, `script.Args`), then we don't need this change. I don't think anyone's likely to call `IfExists(pathname)` with a pathname with a trailing newline, are they?